overlord/stateengine.go

```diff
@@ -35,6 +35,9 @@ type StateManager interface {
 	// See StateEngine.Ensure for more details.
 	Ensure() error
 
+	// Wait asks manager to wait for all running activities to finish.
+	Wait() error
```

Comment:
This change makes me want to try something: can we drop the error result from both Stop and Wait, until we learn we need them?  This are generally mechanical/operational functions that tell the internals to change their state. It seems better and easier for us to route real errors through either Ensure or the tasks and changes.